### PR TITLE
[1.6] Sanitize render_markdown() output with nh3 library

### PR DESCRIPTION
**Pull request by glennmatthews**
# What's Changed

1.6 backport of #5133. Some code location changes (e.g. `nautobot/utilities` vs `nautobot/core`) but otherwise the same code.

# Screenshots

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/utilities/forms/fields.py [hunk 1]
```diff
@@ -391,12 +393,16 @@ class CommentField(forms.CharField):
...
-        "Markdown</a> syntax is supported"
-    )
+
+    @property
+    def default_helptext(self):
+        # TODO: Port Markdown cheat sheet to internal documentation
+        return format_html(
+            '<i class="mdi mdi-information-outline"></i> '
+            '<a href="https://www.markdownguide.org/cheat-sheet/#basic-syntax" rel="noopener noreferrer">Markdown</a> '
+            'syntax is supported, well as <a href="{}#render_markdown">a limited subset of HTML</a>.',
```
Comment (HanlinMiao) [inline nautobot/utilities/forms/fields.py:403]:
```suggestion
            'syntax is supported, as well as <a href="{}#render_markdown">a limited subset of HTML</a>.',
```

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/utilities/forms/fields.py [hunk 2]
```diff
@@ -391,12 +393,16 @@ class CommentField(forms.CharField):
...
-    )
+
+    @property
+    def default_helptext(self):
+        # TODO: Port Markdown cheat sheet to internal documentation
+        return format_html(
+            '<i class="mdi mdi-information-outline"></i> '
+            '<a href="https://www.markdownguide.org/cheat-sheet/#basic-syntax" rel="noopener noreferrer">Markdown</a> '
+            'syntax is supported, as well as <a href="{}#render_markdown">a limited subset of HTML</a>.',
+            static("docs/user-guide/platform-functionality/template-filters.html"),
```
Comment (glennmatthews) [inline nautobot/utilities/forms/fields.py:404]:
```suggestion
            static("docs/additional-features/template-filters.html"),
```

1.6 vs 2.x differences.

File: nautobot/utilities/templatetags/helpers.py [hunk 3]
```diff
@@ -162,18 +163,13 @@ def render_markdown(value):
...
-    value = re.sub(pattern, "[\\1](\\3)", value, flags=re.IGNORECASE)
-
     # Render Markdown
     html = markdown(value, extensions=["fenced_code", "tables"])
 
-    return mark_safe(html)  # noqa: S308
+    # Sanitize rendered HTML
+    html = clean_html(html)
+
+    return mark_safe(html)  # noqa: S308  # suspicious-mark-safe-usage, OK here since we sanitized the string earlier
```
Comment (gsnider2195) [inline nautobot/utilities/templatetags/helpers.py:172]: Not blocking, but I wonder if `clean_html` should return a safe string?

**Review by gsnider2195 — APPROVED** [review]

File: nautobot/utilities/templatetags/helpers.py (hunk 3 above)
Comment (glennmatthews) [reply nautobot/utilities/templatetags/helpers.py:172]: Maybe so - can revisit this once we have more than a single consumer of `clean_html` to think about?

**Pull request merged**